lib/foundation/networking/dio.dart

```diff
@@ -36,7 +37,8 @@ Dio newDio(
   final baseUrl = args.baseUrl;
   final booruFactory = args.booruFactory;
 
-  final booru = booruFactory.getBooruFromUrl(baseUrl);
+  final booru = booruFactory.getBooruFromUrl(baseUrl) ??
+      booruFactory.getBooruFromId(booruConfig.booruId);
```

Comment:
This is needed because otherwise no booru is selected when coming from the "add new booru" page and an unknown URL is used, even though the user has told which booru it is.

Reply:
It works fine on my end. Are you sure this is an issue?

Reply:
Yes, this is an issue when using boorus that don't have known URLs, in my case I have `https://booru.shiro.lan/`, that I tried to add as Danbooru, and when it tries to test it, `booru` is `null` because it tries to get the booru type from the URL, not from what was selected in the dropdown.

Reply:
Please merge the latest changes from dev and see if the issue is still there.

Reply:
The issue doesn't manifest anymore because I removed the initial change that only used the native adapter on boorus that had their protocol set as `native`.

It might not cause any issues now since the native adapter is used for everything, but I'd like to prevent this from causing any issues in the future if possible.